### PR TITLE
Tls

### PR DESCRIPTION
**Pull request by skothari-tibco**
Add TLS/Mutual SSL support

### CODE REVIEW

**Comment by mellistibco** [comment]
@skothari-tibco Can you remove all the other test files that have been changed? It would be better to just have the changes for this specific PR :)

**Comment by skothari-tibco** [comment]
All the changes to other files are due to `go fmt`. Would that be ok?

**Comment by mellistibco** [comment]
Would prefer to not touch a file that doesn't have any significance to this pr...

**Comment by skothari-tibco** [comment]
This PR can be merged after #27

**Comment by fm-tibco** [comment]
@skothari-tibco  #27, was merged, can you cleanup this PR

**Comment by skothari-tibco** [comment]
It's ready for merge.

File: trigger/rest/descriptor.json [hunk 1]
```diff
@@ -10,6 +10,19 @@
       "name": "port",
       "type": "int",
       "required": true
+    },
+    {
+      "name":"tls",
+      "type":"boolean"
+    },
+    {
+      "name": "certs",
```
Comment (fm-tibco) [inline trigger/rest/descriptor.json]: Your descriptor and metadata do not match.  We should definitely come up with a dev CLI plugin to validate a plugin.  In either case should't the both be something more descriptive like keyPemFile and certPemFile?  As for tls should that be "useTLS" maybe?

File: trigger/rest/metadata.go [hunk 2]
```diff
@@ -5,7 +5,10 @@ import (
 )
 
 type Settings struct {
-	Port int `md:"port,required"`
+	Port   int    `md:"port,required"`
+	TLS    bool   `md:"tls"`
+	CertPm string `md:"certpm"`
```
Comment (fm-tibco) [inline trigger/rest/metadata.go]: should match descriptor,

File: trigger/cli/utils.go [hunk 3]
```diff
@@ -98,7 +98,6 @@ type mainCmdUsageData struct {
 
 var mainUsageTpl = `Usage:
```
Comment (fm-tibco) [inline trigger/cli/utils.go]: Remove this file from this PR

File: activity/rest/descriptor.json [hunk 4]
```diff
@@ -29,6 +29,24 @@
       "name": "skipSSL",
       "type": "boolean",
       "value": "false"
+    },
+    {
+      "name":"useTLS",
```
Comment (fm-tibco) [inline activity/rest/descriptor.json]: remove useTLS

File: activity/rest/metadata.go [hunk 5]
```diff
@@ -5,12 +5,16 @@ import (
...
-	Proxy   string            `md:"proxy"`
-	SkipSSL bool              `md:"skipSSL"`
-	Timeout int               `md:"timeout"`
+	Method        string            `md:"method,required,allowed(GET,POST,PUT,PATCH,DELETE)"`
+	Uri           string            `md:"uri,required"`
+	Headers       map[string]string `md:"headers"`
+	Proxy         string            `md:"proxy"`
+	SkipSSLVerify bool              `md:"skipSSL"`
+	Timeout       int               `md:"timeout"`
+	TLS           bool              `md:"useTLS"`
```
Comment (fm-tibco) [inline activity/rest/metadata.go]: remove TLS

**Pull request merged**